jberet-core/src/main/java/org/jberet/repository/JdbcRepository.java

```diff
@@ -1061,11 +1067,20 @@ public void executeStatements(final String statements, final String statementsRe
             connection = getConnection();
             preparedStatement = connection.prepareStatement(selectSql);
             preparedStatement.setString(1, jobName);
+            BatchLogger.LOGGER.debugf("Executing query to load job executions: %s", selectSql);
             rs = preparedStatement.executeQuery();
+            BatchLogger.LOGGER.debugf("Reading job execution records");
+            int count = 0;
             while (rs.next()) {
                 final long i = rs.getLong(1);
                 result.add(i);
+                count++;
+                if (maxRecords != null && count >= maxRecords) {
+                    BatchLogger.LOGGER.jobExecutionRecordsLimited(maxRecords);
+                    break;
+                }
             }
+            BatchLogger.LOGGER.debugf("Number of job execution records read: %d", result.size());
```

Comment:
can we add this into a separate loop for a non-null `maxRecords`?  Most users probably won't bother to configure this property. Something like:

```java
if(null) {
  //old while loop
  while (...) {...}
} else {
 //new loop
 count = 0;
 while(rs.next() && count < maxRecords) {
    ...
 }
 log(...)
}

Reply:
Sure, I updated the PR.